src/Microsoft.NetCore.Analyzers/Core/Runtime/TestForEmptyStringsUsingStringLength.Fixer.cs

```diff
@@ -18,12 +25,139 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+
+            SyntaxNode binaryExpressionSyntax = GetBinaryExpression(node);
+
+            if (!IsEqualsOperator(binaryExpressionSyntax) && !IsNotEqualsOperator(binaryExpressionSyntax))
+            {
+                return;
+            }
+
+            SemanticModel model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            FixResolution resolution = TryGetFixResolution(binaryExpressionSyntax, model);
+
+            if (resolution != null)
+            {
+                // We cannot have multiple overlapping diagnostics of this id.
+                Diagnostic diagnostic = context.Diagnostics.Single();
+
+                var methodInvocationAction = CodeAction.Create(SystemRuntimeAnalyzersResources.TestForEmptyStringsUsingStringLengthMessage,
+                    async ct => await ConvertToMethodInvocation(context, resolution).ConfigureAwait(false),
+                    equivalenceKey: "TestForEmptyStringCorrectlyUsingIsNullOrEmpty");
+
+                context.RegisterCodeFix(methodInvocationAction, diagnostic);
+
+                var stringLengthAction = CodeAction.Create(SystemRuntimeAnalyzersResources.TestForEmptyStringsUsingStringLengthMessage,
+                    async ct => await ConvertToStringLengthComparison(context, resolution).ConfigureAwait(false),
+                    equivalenceKey: "TestForEmptyStringCorrectlyUsingStringLength");
+
+                context.RegisterCodeFix(stringLengthAction, diagnostic);
+            }
+        }
+
+        private FixResolution TryGetFixResolution(SyntaxNode binaryExpressionSyntax, SemanticModel model)
+        {
+            bool isEqualsOperator = IsEqualsOperator(binaryExpressionSyntax);
+            SyntaxNode leftOperand = GetLeftOperand(binaryExpressionSyntax);
+            SyntaxNode rightOperand = GetRightOperand(binaryExpressionSyntax);
+
+            if (ContainsSystemStringEmpty(leftOperand, model))
+            {
+                return new FixResolution(binaryExpressionSyntax, rightOperand, isEqualsOperator);
+            }
+
+            if (ContainsSystemStringEmpty(rightOperand, model))
+            {
+                return new FixResolution(binaryExpressionSyntax, leftOperand, isEqualsOperator);
+            }
+
+            return null;
+        }
+
+        private static bool ContainsSystemStringEmpty(SyntaxNode expressionSyntax, SemanticModel model)
+        {
+            if (model.GetSymbolInfo(expressionSyntax).Symbol is IFieldSymbol fieldSymbol)
+            {
+                if (fieldSymbol.Type.Equals(model.Compilation.GetTypeByMetadataName("System.String")))
```

Comment:
Better approach is to compare `fieldSymbol.Type.SpecialType` is the special type for string.

Reply:
thanks!